go-controller/pkg/node/linkmanager/link_network_manager.go

```diff
@@ -0,0 +1,389 @@
+		if !found {
+			// we dont managed this link, check we  used to manage it and if a stale address is present, delete it
+			for _, address := range addressesFound {
+				// we label any address we create, so if we aren't managed a link, we must remove any stale addresses
+				if address.Label == GetAssignedAddressLabel(link.Attrs().Name) {
+					if err := netlinkOps.AddrDel(link, &address); err != nil {
+						klog.Errorf("Link Network Manager: failed to delete address %q from link %q",
+							address.String(), link.Attrs().Name)
+					} else {
+						klog.Infof("Link Network Manager: successfully removed stale address %q from link %q",
+							address.String(), link.Attrs().Name)
+					}
+				}
+			}
+			continue
+		}
+		// ensure any addresses found that we previously managed are deleted
+		for _, addrFound := range addressesFound {
+			if !containsAddress(addressesWanted, addrFound) {
+				// every address we added is labeled with a well-known label to ensure we clean any addresses we previously managed.
+				if addrFound.Label == GetAssignedAddressLabel(link.Attrs().Name) {
+					// delete an unmanaged address that we used to managed
+					if err = netlinkOps.AddrDel(link, &addrFound); err != nil {
+						klog.Errorf("Link manager: failed to delete stale address %q from link %q: %v",
+							addrFound, link.Attrs().Name, err)
+					} else {
+						klog.Infof("Link manager: deleted stale assigned address %s on link %s", addrFound.String(), link.Attrs().Name)
+					}
+
+				}
+			}
+		}
+		// add the addresses we want that are not found on the link
+		for _, addressWanted := range addressesWanted {
+			var exists bool
+			for _, addressFound := range addressesFound {
+				if addressFound.Equal(addressWanted) {
+					exists = true
+					break
+				}
+			}
+			if !exists {
+				// Use arping to try to update other hosts ARP caches, in case this IP was
+				// previously active on another node. (Based on code from "ifup".)
+				go func() {
+					out, err := exec.Command("/sbin/arping", "-q", "-A", "-c", "1", "-I", link.Attrs().Name, addressWanted.IP.String()).CombinedOutput()
+					if err != nil {
+						klog.Warningf("Failed to send ARP claim for IP %s: %v (exec command output: %q)", addressWanted.IP.String(), err, string(out))
+						return
+					}
+					time.Sleep(2 * time.Second)
+					_ = exec.Command("/sbin/arping", "-q", "-U", "-c", "1", "-I", link.Attrs().Name, addressWanted.IP.String()).Run()
+				}()
+
+				if err = netlinkOps.AddrAdd(link, &addressWanted); err != nil {
+					klog.Errorf("Link manager: failed to add address %q to link %q: %v",
+						addressWanted.String(), link.Attrs().Name, err)
+				}
+				// Use arping to try to update other hosts ARP caches, in case this IP was
+				// previously active on another node. (Based on code from "ifup".)
+				go func() {
+					out, err := exec.Command("/sbin/arping", "-q", "-A", "-c", "1", "-I", link.Attrs().Name, addressWanted.IP.String()).CombinedOutput()
```

Comment:
did you consider using the golang arping package?

https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/net_linux.go#L367

Reply:
Nope. I didnt know this existed. Want me to switch?

Reply:
I copied directly from SDN.

Reply:
Switched to the arping package.